ansible/playbooks/roles/ocp-config/tasks/main.yaml

```diff
@@ -1,5 +1,54 @@
 ---
 # tasks file for ocp4 config
+- name: Extract OCP4 tools
+  when: release_image_override != ""
+  block:
+    - name: Set tools directory path
+      set_fact:
+        tools_dir: "~/ocp4_tools"
+
+    - name: Create tools directory
+      file:
+        path: "{{ tools_dir }}"
+        state: "{{ item }}"
+        mode: '0755'
+      with_items:
+      - absent
+      - directory
+
+    - name: Extract OCP4 tools from release image
+      shell: |
+        echo '{{ install_config.pull_secret }}' > pull_secret
```

Comment:
Is this step really needed? Can the pull_secret input be directly used without copying it to another file ?

Reply:
The `oc adm` command parameter expects a path to pull secret file `--registry-config='': Path to your registry credentials` hence above approach